### PR TITLE
Use `GenStructStep` from `structural_2` in `DeriveSha`.

### PR DESCRIPTION
**Pull request by mandrigin**
fixes: #148

### CODE REVIEW

File: core/types/derive_sha.go [hunk 1]
```diff
@@ -17,25 +17,141 @@
...
 
 import (
-	"bytes"
-
 	"github.com/ledgerwatch/turbo-geth/common"
 	"github.com/ledgerwatch/turbo-geth/rlp"
 	"github.com/ledgerwatch/turbo-geth/trie"
 )
 
+var emptyHash = trie.New(common.Hash{}).Hash()
```
Comment (AlexeyAkhunov) [inline core/types/derive_sha.go]: Does it end up being the same as this one: https://github.com/ledgerwatch/turbo-geth/blob/28389a1cb40ffd2640f323c780538e5ca8de7810/trie/trie.go#L33 ?

Comment (mandrigin) [reply core/types/derive_sha.go]: ah, it's a constant 😆
yeah, it is the same, and I'll fix the PR to actually use this constant.

Comment (mandrigin) [reply core/types/derive_sha.go]: fixed

File: core/types/derive_sha.go [hunk 2]
```diff
@@ -30,12 +28,128 @@ type DerivableList interface {
...
+	curr := &trie.OneBytesTape{}
+	succ := &trie.OneBytesTape{}
+	value := &trie.OneBytesTape{}
+
+	hb := trie.NewHashBuilder()
+
+	hb.SetKeyTape(curr)
+	hb.SetValueTape(value)
+
+	hb.Reset()
```
Comment (mandrigin) [inline core/types/derive_sha.go]: remove

**Pull request merged**